hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -175,13 +175,19 @@ public BackgroundTaskQueue getTasks() {
   public List<ContainerBlockInfo> chooseContainerForBlockDeletion(
       int blockLimit, ContainerDeletionChoosingPolicy deletionPolicy)
       throws StorageContainerException {
+    long[] totalPendingBlockCount = new long[1];
     Map<Long, ContainerData> containerDataMap =
         ozoneContainer.getContainerSet().getContainerMap().entrySet().stream()
             .filter(e -> ((KeyValueContainerData) e.getValue()
                 .getContainerData()).getNumPendingDeletionBlocks() > 0)
             .filter(e -> isDeletionAllowed(e.getValue().getContainerData(),
-                deletionPolicy)).collect(Collectors
+                deletionPolicy))
+            .peek(e -> totalPendingBlockCount[0] +=
```

Comment:
I'm not sure `peek` is good to use here.

An explicit sum on the result map might be better.

Reply:
`peek` seems not a stable api to use for now according to its description.

Clearer code can be better for review.

Reply:
Thanks for the update. LGTM.